test/one_dragon/launcher/test_src_recovery.py

```
@@ -0,0 +1,301 @@
+        assert not src.exists()
+
+    def test_backup_missing_src(self, tmp_path: Path) -> None:
+        src = tmp_path / 'src'
+        backup_dir, error = _backup_src_dir(src)
+        assert error == ''
+        assert backup_dir is None
+
+    def test_backup_unique_name(self, tmp_path: Path) -> None:
+        src = tmp_path / 'src'
+        src.mkdir()
+        # 制造一个同名前缀目录，验证重名时自动加后缀
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)